release.js

```diff
@@ -374,46 +376,107 @@ function createWebsiteDirectory() {
 
 
 function createDistDirectory() {
-  fs.mkdir(dir + '/dist', function(err) {
+  var distdir = dir + '/dist';
+  fs.mkdir(distdir, function(err) {
     if (err && err.code !== 'EEXIST')
-        return pauseRetry(err);
+      return pauseRetry(err);
+    if (fs.readdirSync(distdir).length !== 0)
+      return pauseRetry("dist dir is not empty");
 
     next();
   });
 }
 
 
-function createTarBall() {
+function createCMakeTarBall() {
   var tag = ver.format(state.releaseVersion);
   var filename = format('libuv-%s.tar.gz', tag);
   var prefix = format('libuv-%s/', tag);
   var command = format('git archive %s --format=tar --prefix=%s | gzip -9 > dist/%s',
                        tag,
                        prefix,
                        filename);
-  child_process.exec(command, { stdio: 'inherit', cwd: dir}, nextOrRetry);
+  child_process.exec(command, { stdio: 'inherit', cwd: dir }, nextOrRetry);
 }
 
 
-function signTarball() {
+function createAutotoolsTarBall() {
   var tag = ver.format(state.releaseVersion);
-  var filename = format('libuv-%s.tar.gz', tag);
-  var signFilename = format('libuv-%s.tar.gz.sign', tag);
-  var command = format('gpg -a --output dist/%s --detach-sign dist/%s',
-                       signFilename,
-                       filename);
-  child_process.exec(command, { stdio: 'inherit', cwd: dir}, nextOrRetry);
+  try {
+  var distver = format('%d.%d.%d',
+                       state.releaseVersion.major,
+                       state.releaseVersion.minor,
+                       state.releaseVersion.patch);
+  } catch (err) {
+    return pauseRetry(err);
+  }
```

Comment:
Superfluous try/catch? Can't fail, right? The indentation inside the try block is off, too.

Reply:
huh, how did this work? I seem to have given the distver local scope here, so I couldn't use it later. I think I had stuck this in while developing since I was annoyed that the script aborted the release if there was any errors thrown, and some things are slow or not fully idempotent. Removed.